lhotse/supervision.py

```diff
@@ -75,6 +75,15 @@ def filter(self, predicate: Callable[[SupervisionSegment], bool]) -> 'Supervisio
         """
         return SupervisionSet.from_segments(seg for seg in self if predicate(seg))
 
+    def modify(self, process: Callable[[SupervisionSegment], None]) -> None:
```

Comment:
Suggestions:
- similar changes as in the `CutSet`
- rename `modify` to `map`
- rename `process` to `transform_fn` with signature `Callable[[SupervisionSegment], SupervisionSegment]`
- return a new `SupervisionSet` (`SupervisionSet.from_segments(...)`)